File: compiler/server/format.go
package server

import (
	"wipple/syntax"
)

type FormatRequest struct {
	Code string `json:"code"`
}

type FormatResponse struct {
	Code string `json:"code"`
}

func (request *FormatRequest) handle() (FormatResponse, error) {
	formatted, err := syntax.Format(request.Code)
	if err != nil {
		// Return the original code if it can't be parsed
		return FormatResponse{Code: request.Code}, nil
	}

	return FormatResponse{Code: formatted}, nil
}

File: compiler/server/format.rs
use serde::Deserialize;
use serde_json::json;
use wipple::syntax::format;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Request {
    code: String,
}

pub async fn handle(request: Request) -> anyhow::Result<serde_json::Value> {
    let formatted = format(&request.code).filter(|formatted| *formatted != request.code);

    Ok(json!({ "code": formatted }))
}
